src/shadowbox/server/manager_metrics.ts

```diff
@@ -40,4 +72,71 @@ export class PrometheusManagerMetrics implements ManagerMetrics {
     }
     return {bytesTransferredByUserId: usage};
   }
+
+  async getServerMetrics({hours}: ServerMetricsTimeframe): Promise<ServerMetrics> {
+    const dataTransferredByLocation = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_data_bytes_per_location{dir=~"c<p|p>t"}[${hours}h])) by (location, asn, asorg)`
+    );
+    const tunnelTimeByLocation = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_tunnel_time_seconds_per_location[${hours}h])) by (location, asn, asorg)`
+    );
+    const dataTransferredByAccessKey = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_data_bytes{dir=~"c<p|p>t"}[${hours}h])) by (access_key)`
+    );
+    const tunnelTimeByAccessKey = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_tunnel_time_seconds[${hours}h])) by (access_key)`
+    );
+
+    const tunnelTimeByLocationMap = new Map();
+    for (const entry of tunnelTimeByLocation.result) {
+      tunnelTimeByLocationMap.set(
+        `${entry.metric['location']},${entry.metric['asn']},${entry.metric['asorg']}`,
+        parseFloat(entry.value[1])
+      );
+    }
+
+    const server = [];
+    for (const entry of dataTransferredByLocation.result) {
```

Comment:
If a key is in tunnelTimeByLocation, but not in dataTransferredByLocation, you won't output it.
You need to make sure you output the union.
Let's add a test case that exercises that.